### PR TITLE
Plugins: Expose backendplugin for client proto interface

### PR DESCRIPTION
**Pull request by wbrowne**
**What is this feature?**

Adds some more methods to the plugin client proto interface.

**Why do we need this feature?**

This makes our life easier when working with a proto client as it can use both proto and SDK contracts (making it play nicer with existing plugin management components).

**Special notes for your reviewer:**
https://github.com/grafana/grafana-enterprise/pull/6492

### CODE REVIEW

File: pkg/registry/apis/datasource/handler.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	"context"
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+type HTTPRequestHandlerFunc func(ctx context.Context, pluginCtx backend.PluginContext, responder rest.Responder) (http.Handler, error)
+
+type PluginRequestHandlers struct {
```
Comment (ryantxu) [inline pkg/registry/apis/datasource/handler.go:13]: 🤔 why not https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/datasource/serve.go#L8 

IIUC, it achieves the same thing -- and keeps the core vs external distinction irrelevant for anyone using it.

Comment (wbrowne) [reply pkg/registry/apis/datasource/handler.go:13]: If we use the SDK contracts here, then for a response for an external plugin over gRPC will go:

`[plugin gRPC server]` -> encode as Arrow (backend to proto)  -> `[plugin gRPC client]` -> decode Arrow (proto to backend) -> `[gRPC server]` -> encode as Arrow (backend to proto) -> `[gRPC client]` -> decode Arrow (proto to backend)

As opposed to:

`[plugin gRPC server]` -> encode as Arrow (backend to proto) -> `[plugin gRPC client]` -> passthru -> `[gRPC server]` -> passthru -> `[gRPC client]` -> decode Arrow (proto to backend)

Comment (ryantxu) [reply pkg/registry/apis/datasource/handler.go:13]: I understand that for the protobuf side... but not for HTTP

Comment (wbrowne) [reply pkg/registry/apis/datasource/handler.go:13]: Yes exactly - I made a note in the description that this doesn't need to be this way for HTTP, but for now it seemed more straightforward to consolidate both paths into one. If that's a deal breaker, then we can look at trying to keep the structure as is but throwing the client proto stuff in on top?

File: go.mod [hunk 2]
```diff
@@ -51,7 +51,7 @@ require (
 	github.com/grafana/cuetsy v0.1.11 // @grafana/grafana-as-code
 	github.com/grafana/grafana-aws-sdk v0.25.0 // @grafana/aws-datasources
 	github.com/grafana/grafana-azure-sdk-go/v2 v2.0.1 // @grafana/partner-datasources
-	github.com/grafana/grafana-plugin-sdk-go v0.222.0 // @grafana/plugins-platform-backend
+	github.com/grafana/grafana-plugin-sdk-go v0.223.1-0.20240416132542-5d8bc8d25317 // @grafana/plugins-platform-backend
```
Comment (wbrowne) [inline go.mod:54]: TODO need to actually open a PR for this one https://github.com/grafana/grafana-plugin-sdk-go/compare/main...expose-serve-opts

**Review by ryantxu — APPROVED** [review]

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**